scipy/stats/_hypotests.py

```diff
@@ -956,6 +958,204 @@ def barnard_exact(table, alternative="two-sided", pooled=True, n=32):
     return BarnardExactResult(wald_stat_obs, p_value)
 
 
+BoschlooExactResult = make_dataclass(
+    "BoschlooExactResult", [("statistic", float), ("pvalue", float)]
+)
+
+
+def boschloo_exact(table, alternative="two-sided", n=32):
+    r"""Perform a Boschloo exact test on a 2x2 contingency table.
+
+    Parameters
+    ----------
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the null and alternative hypotheses. Default is 'two-sided'.
+        Please see explanations in the Notes section below.
+
+    n : int, optional
+        Number of sampling points used in the construction of the sampling
+        method. Note that this argument will automatically be converted to
+        the next higher power of 2 since `scipy.stats.qmc.Sobol` is used to
+        select sample points. Default is 32. Must be positive. In most cases,
+        32 points is enough to reach good precision. More points comes at
+        performance cost.
+
+    Returns
+    -------
+    ber : BoschlooExactResult
+        A result object with the following attributes.
+
+        statistic : float
+            The Fisher statistic.
+
+        pvalue : float
+            The probability of obtaining a test statistic at least as extreme
+            as the one observed under the null hypothesis.
+
+    See Also
+    --------
+    chi2_contingency : Chi-square test of independence of variables in a
+        contingency table.
+    fisher_exact : Fisher exact test on a 2x2 contingency table.
+    barnard_exact : Barnard's exact test, which is a more powerful alternative
+        than Fisher's exact test for 2x2 contingency tables.
+
+    Notes
+    -----
+    Boschloo's test is an exact test used in the analysis of contingency
+    tables. It examines the association of two categorical variables, and
+    is a uniformly more powerful alternative to Fisher's exact test
+    for 2x2 contingency tables.
+
+    Let's define :math:`X_0` a 2x2 matrix representing the observed sample,
+    where each column stores the binomial experiment, as in the example
+    below. Let's also define :math:`p_1, p_2` the theoretical binomial
+    probabilities for  :math:`x_{11}` and :math:`x_{12}`. When using
+    Boschloo exact test, we can assert three different null hypotheses :
+
+    - :math:`H_0 : p_1 \geq p_2` versus :math:`H_1 : p_1 < p_2`,
+      with `alternative` = "less"
+
+    - :math:`H_0 : p_1 \leq p_2` versus :math:`H_1 : p_1 > p_2`,
+      with `alternative` = "greater"
+
+    - :math:`H_0 : p_1 = p_2` versus :math:`H_1 : p_1 \neq p_2`,
+      with `alternative` = "two-sided" (default one)
+
+    In order to compute Boschloo's exact test, we are using the Fisher's
+    pvalue as a statistic.
+
+    Boschloo's and Barnard's are both more powerful than Fisher's exact
+    test. From some local testing, it appears that Boschloo is slightly
```

Comment:
If we're comparing them, should probably mention somewhere the conditions under which one is more appropriate. My understanding is that only Fisher's would be appropriate if the row sums are truly fixed, as in the lady tasting tea experiment, and Barnard's/Boschloo's would be more appropriate if the row sums are not fixed a priori (which is more common, probably).

Reply:
Yes Fisher is appropriate if both margins are fixed (rows and columns).

As for this sentence, I was surprised as the articles I read indicate that Barnard would be more exact and faster 🤷‍♂️ But I cannot find a proper review of both methods. @jb-leger would you know?

Side note: if we can calculate the power, could we report it @V0lantis?

Reply:
For my comment where I calculate the power of the test, I use a binomial distribution with a fixed and known proportion proportion for `p_1` and `p_2`. The power can only be calculated if we now either the true probability, or if we assert them. 

We could add this functionality in an other PR if you wish, where a power would be given for given value of `p_1` and `p_2`

Reply:
This comment will not be blocking, but I would still suggest removing comments about informal tests and only make statements like this if they're supported by extensive tests or we can cite literature.

Reply:
Removed in 9a3b92bde3015c2214f95433914fea95cc4b309b. 
If I write some blog post with extensive testing, will I be able to link the blog post inside this doc?